MainWindow.cpp

```diff
@@ -468,6 +468,28 @@ void MainWindow::on_actionOpen_triggered() {
   if (!fileName.isEmpty()) openFile(fileName);
 }
 
+void MainWindow::on_actionSave_triggered() {
+  // map useful to quickly fetch extension from the selected filter in QDialog (because Qt doesn't give a better way)
+  QHash<QString, QString> extensionMap;
+  extensionMap["EGM (*.egm)"] = ".egm";
+
+  QString selectedFilter;
+  QString fileName = QFileDialog::getSaveFileName(this, tr("Save Project"), "",
+                                                  extensionMap.keys().join(QStringLiteral(";;")),
+                                                  &selectedFilter);
+
+  if(!fileName.endsWith(extensionMap[selectedFilter], Qt::CaseInsensitive)) {
+    // removes any trailing periods(.) from the file path
+    while(fileName.endsWith(QLatin1Char('.')))
+      fileName.chop(1);
+    fileName.append(extensionMap[selectedFilter]);
+  }
+  std::cout<<fileName.toStdString()<<"\n";
+  if(!fileName.isEmpty()) {

```

Comment:
This check needs to be moved up. Now it will never be false, since we'd have added the extension.